modules/LaneDetection/src/LaneManager.cpp

```diff
@@ -9,40 +9,45 @@ namespace XyCar
 
     void LaneManager::image_callback(const sensor_msgs::Image& message)
     {
-        image_ = cv::Mat(message.height, message.width, CV_8UC3, const_cast<uint8_t*>(&message.data[0]), message.step);
-        cv::cvtColor(image_, image_, cv::COLOR_RGB2BGR);
-        if(!image_.empty()){
-            cv::imshow("image", image);
-            cv::waitKey(1);
-        }
-        // current_images_.push(image_processor_.process(image));
+        cv::Mat camera_image = cv::Mat(message.height, message.width, CV_8UC3, const_cast<uint8_t*>(&message.data[0]), message.step);
+        cv::cvtColor(camera_image, image_, cv::COLOR_RGB2BGR);
     }
 
     void LaneManager::run()
     {
-        // if (current_images_.empty())
-        //     return;
-
-        // cv::Mat current_image = current_images_.front();
-        // current_images_.pop();
-`
-        if(!image_.empty()){
-            //detect rpos, lpos, and flag of stop
-            std::tuple<int32_t, int32_t, bool> output_detector;
+        ros::Rate rate(33);
+        while (ros::ok())
+        {
+            ros::spinOnce();
+            std::cout << image_.size() << std::endl;
+
+            if(image_.empty())
+                continue;
+
             cv::Mat canny_image = image_processor_.process(image_);
-            output_detector = detector_.findPos(current_image);
+
+            std::tuple<int32_t, int32_t, bool> output_detector;
+            output_detector = detector_.findPos(canny_image);
             int32_t left_pos = std::get<0>(output_detector);
             int32_t right_pos = std::get<1>(output_detector);
             bool is_stop = std::get<2>(output_detector);
 
+            // std::cout << "lpos: " << left_pos << std::endl;
+            // std::cout << "rpos: " << right_pos << std::endl;
+            // std::cout << "stop: " << is_stop << std::endl;
+
             int32_t error = k_frame_width / 2 - static_cast<int32_t>((right_pos + left_pos) / 2);
 
             PREC angle = pid_controller_.computeAngle(error);
+            // std::cout <<"angle: " << angle << std::endl;
 
             xycar_controller.control(angle);
-        }
 
-        
+            cv::imshow("image_", image_);
+            cv::imshow("canny_image", canny_image);
+
+            cv::waitKey(1);
```

Comment:
```suggestion

            if (!DEBUG)
                continue;
            
            cv::imshow("image_", image_);
            cv::imshow("canny_image", canny_image);

            cv::waitKey(1);
```
이후에 configuration으로 DEBUG값을 설정해준다면 그 값에 따라 손쉽게 이미지를 디버깅 할 수도 있겠습니다

Reply:
아 이미 cmake에 추가하셨군요. 그럼 전처리기를 활용해보시길 바랍니다 :)